assignment3.txt

```diff
@@ -0,0 +1,10 @@
+Christopher Liu
+
+1. No, there is no input for the formatted string. If there were, you would have to assume that i was declared as a positive integer. If a float, the loop would not work as it takes integers, and if it is negative, it would last forever.
+
+2. The loop is going to be set to 0 since when it iterates, it will go to -1 then 0.5 then truncate to 0, resulting in an infinite setting to 0. There is also a semicolon after the for loop parameters, which would make the loop do nothing. 
+
+3. i. O(n^2)
+ii. O(n^3)
```

Comment:
The answer is actually O(n^4) - again, for loops that run from 0 to n have the time complexity O(n). So the top loop is O(n), the middle Loop is O(n^2) iterations because it iterates from 1 to i^2. The if statement is O(1 / n) times because of the modulo function. Lastly, the inner loop  is O(n^2) iterations because it iterates from 0 to j and j = i^2 = n^2. 

Therefore, the sum = -(1/12)*n + (3/8)*n^2 - (5/12)*n^3 + (1/8)*n^4  = O(n^4). You should explain your answers so we know where you went wrong.